pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/SegmentPushUtils.java

```diff
@@ -371,4 +372,35 @@ private static File generateSegmentMetadataFile(PinotFS fileSystem, URI tarFileU
       FileUtils.deleteQuietly(segmentMetadataDir);
     }
   }
+
+  public static List<String> getSegmentNames(BatchConfigProperties.SegmentPushType pushMode,
```

Comment:
Why do we need to handle this separately for each `pushMode`? I thought `Map<String, String> segmentsUriToTarPathMap` should be in the same format regardless the pushMode. I thought that we can use the same code and put it to the Base class. Am I missing some detail here?

Reply:
Thanks @snleee! As discussed offline, after looking at `SegmentPushUtils.java#getSegmentUriToTarPathMap` again, this separation based on `pushMode` is indeed unnecessary as the URI names are derived from tarFilePaths from `segmentsUriToTarPathMap.values()` anyways, which means TAR, METADATA and URI can be unified.

Hence, removed the overrides and brought generic implementation up to the base class.